doc/templates.rst

```diff
@@ -784,7 +784,10 @@ The following operators don't fit into any of the other categories:
 
 * ``~``: Converts all operands into strings and concatenates them. ``{{ "Hello
   " ~ name ~ "!" }}`` would return (assuming ``name`` is ``'John'``) ``Hello
-  John!``.
+  John!``. Note that the output will be escaped due to autoescaping.
+  
+  If you wish to have raw output, group the concatenation operands in parentheses 
+  before passing them to the raw filter: ``{{ ("Hello " ~ name ~ "!")|raw }}``
```

Comment:
I don't think that this is the right place. It's more about operator precedence, where we already have the information. I've tried to make it clearer in #2802